src/Spice86.Core/Emulator/InterruptHandlers/Dos/Xms/ExtendedMemoryManager.cs

```diff
@@ -0,0 +1,490 @@
+            return;
+        }
+
+        _xmsHandles[handle] = lockCount - 1;
+
+        _state.AX = 1; // Success.
+    }
+
+    /// <summary>
+    /// Returns information about an XMS handle.
+    /// </summary>
+    public void GetHandleInformation() {
+        int handle = _state.DX;
+
+        if (!_xmsHandles.TryGetValue(handle, out int lockCount)) {
+            _state.AX = 0; // Didn't work.
+            _state.BL = 0xA2; // Invalid handle.
+            return;
+        }
+
+        _state.BH = (byte)lockCount;
+        _state.BL = (byte)(MaxHandles - _xmsHandles.Count);
+
+        if (!TryGetBlock(handle, out XmsBlock? block)) {
+            _state.DX = 0;
+        } else {
+            _state.DX = (ushort)(block.Length / 1024u);
+        }
+
+        _state.AX = 1; // Success.
+    }
+
+    /// <summary>
+    /// Copies a block of memory.
+    /// TODO: Verify this works
+    /// </summary>
+    public unsafe void MoveExtendedMemoryBlock() {
+        bool a20State = _memory.A20Gate.IsEnabled;
+        _memory.A20Gate.IsEnabled = true;
+
+        XmsMoveData moveData = new XmsMoveData(_memory, _state.DS);
+        uint srcAddress = 0;
+        uint destAddress = 0;
+
+        if (moveData.SourceHandle == 0) {
+            srcAddress = MemoryUtils.ToPhysicalAddress(moveData.SourceAddress.Segment, moveData.SourceAddress.Offset);
+        } else {
+            if (TryGetBlock(moveData.SourceHandle, out XmsBlock? srcBlock)) {
+                srcAddress = XmsBaseAddress + srcBlock.Offset + moveData.SourceOffset;
+            }
+        }
+
+        if (moveData.DestHandle == 0) {
+            destAddress = MemoryUtils.ToPhysicalAddress(moveData.DestAddress.Segment, moveData.DestAddress.Offset);
+        } else {
+            if (TryGetBlock(moveData.DestHandle, out XmsBlock? destBlock)) {
+                destAddress = XmsBaseAddress + destBlock.Offset + moveData.DestOffset;
+            }
+        }
+
+        if (srcAddress == 0) {
+            _state.BL = 0xA3; // Invalid source handle.
+            _state.AX = 0; // Didn't work.
+            return;
+        }
+
+        if (destAddress == 0) {
+            _state.BL = 0xA5; // Invalid destination handle.
+            _state.AX = 0; // Didn't work.
+            return;
+        }
+
+        _memory.MemCopy(srcAddress, destAddress, moveData.Length);
+
+        _state.AX = 1; // Success.
+        _memory.A20Gate.IsEnabled = a20State;
+    }
+
+    /// <summary>
+    /// Queries free memory using 32-bit registers.
+    /// </summary>
+    public void QueryAnyFreeExtendedMemory() {
+        _state.EAX = LargestFreeBlock / 1024u;
+        _state.ECX = _memory.Length - 1;
+        _state.EDX = (uint)(TotalFreeMemory / 1024);
+
+        if (_state.EAX == 0) {
+            _state.BL = 0xA0;
+        } else {
+            _state.BL = 0;
+        }
+    }
+
+    public uint Size => XmsMemorySize;
+    
+    public byte Read(uint address) {
+        return _memory[address - XmsBaseAddress];
```

Comment:
Why do you need to store this in emulated memory? This is outside real mode address space so it will never be accessed by programs isnt it?